newIDE/app/src/Utils/GDevelopServices/Shop.js

```diff
@@ -31,3 +31,36 @@ export const listListedPrivateAssetPacks = async (): Promise<
   const response = await client.get('/asset-pack');
   return response.data;
 };
+
+export const getAuthorizationTokenForPrivateAssets = async (
+  getAuthorizationHeader: () => Promise<string>,
+  {
+    userId,
+  }: {|
+    userId: string,
+  |}
+): Promise<string> => {
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await client.post(
+    '/asset-pack/action/authorize',
+    {},
+    {
+      headers: { Authorization: authorizationHeader },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+export const createProductAuthorizedUrl = (
+  url: string,
+  token: string
+): string => {
+  return url + '?token=' + encodeURIComponent(token);
```

Comment:
Should we put a check somewhere that there is not already a `?` in the URL? It shouldn't happen but I'd feel more secure if there was such a thing